python/pylibcudf/pylibcudf/tests/conftest.py

```diff
@@ -20,9 +20,9 @@
 
 def _type_to_str(typ):
     if isinstance(typ, pa.ListType):
-        return f"list-{_type_to_str(typ.value_type)}"
+        return f"list[{_type_to_str(typ.value_type)}]"
     elif isinstance(typ, pa.StructType):
-        return f"struct-{'-'.join([_type_to_str(typ.field(i).type) for i in range(typ.num_fields)])}"
+        return f"struct[{', '.join([_type_to_str(typ.field(i).type) for i in range(typ.num_fields)])}]"
```

Comment:
Passing an iterator should work here afaik. No need to materialize a list.
```suggestion
        return f"struct[{', '.join(_type_to_str(typ.field(i).type) for i in range(typ.num_fields))}]"
```